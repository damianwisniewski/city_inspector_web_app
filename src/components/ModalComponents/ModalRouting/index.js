import ModalLink from './ModalLink/ModalLink'
import ModalRedirect from './ModalRedirect/ModalRedirect'
import ModalRoute from './ModalRoute/ModalRoute'
import ModalRouter from './ModalRouter/ModalRouter'
import ModalSwitch from './ModalSwitch/ModalSwitch'

export { ModalLink, ModalRedirect, ModalRoute, ModalRouter, ModalSwitch }
