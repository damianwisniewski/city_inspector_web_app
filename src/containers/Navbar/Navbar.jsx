import React, { Component } from 'react'
import { connect } from 'react-redux'

import './Navbar.scss'
import iconAssets from '../../assets/styleModules/icons.module.scss'

import logo from '../../assets/images/logo.svg'

import NavItem from '../../components/NavComponents/NavItem/NavItem'
import NavButton from '../../components/NavComponents/NavButton/NavButton'

class Navbar extends Component {
	static defaultProps = {
		isUserAuth: false,
	}

	state = {
		isDrawerOpen: false,
	}

	onClickDrawerToggle = () => this.setState({ isDrawerOpen: !this.state.isDrawerOpen })

	/**
	 * Close drawer after click nav item.
	 * Do not changes drawer state if it's closed (bigger screens)
	 * Do not changes drawer state after click nav item parent
	 */
	onClickMenuOption = e => {
		e.stopPropagation()
		const drawerState = this.state.isDrawerOpen

		if (drawerState && e.target.type) {
			this.setState({ isDrawerOpen: false })
		}
	}

	createAuthNavigation = () => {
		const { isDrawerOpen } = this.state

		return (
			<ul
				id='navList'
				onClick={this.onClickMenuOption}
				className={`navigation_list ${isDrawerOpen ? 'opened' : ''}`}
			>
				<NavItem id='mapLink' exact icon='map' type='link' to='/'>
					Mapa
				</NavItem>
				<NavItem id='plusLink' exact icon='plus' type='link' to='/nowe_zgloszenie'>
					Nowe zgłoszenie
				</NavItem>
				<NavItem id='pinLink' exact icon='pin' type='link' to='/twoje_zgloszenia'>
					Twoje zgłoszenia
				</NavItem>
				<NavItem id='eyeLink' exact icon='eye' type='link' to='/subskrypcje'>
					Subskrypcje
				</NavItem>
				<NavItem id='bellLink' icon='bell' type='button' to='Settings'>
					Powiadomienia
				</NavItem>
				<NavItem id='cogsLink' exact icon='cogs' type='link' to='/ustawienia'>
					Ustawienia
				</NavItem>
				<NavItem id='logoutLink' icon='logout' type='button' to='LogoutPopup'>
					Wyloguj
				</NavItem>
			</ul>
		)
	}

	createNavigation = () => {
		const { isDrawerOpen } = this.state

		return (
			<ul
				id='navList'
				onClick={this.onClickMenuOption}
				className={`navigation_list ${isDrawerOpen ? 'opened' : ''}`}
			>
				<NavItem id='mapLink' exact icon='map' type='link' to='/'>
					Mapa
				</NavItem>
				<NavItem id='helpLink' icon='help' type='button' to='HelpPopup'>
					Pomoc
				</NavItem>
				<NavButton id='registerButton' color='white' to='SignupPopup'>
					Rejestracja
				</NavButton>
				<NavButton id='loginButton' color='blue' to='LoginPopup'>
					Logowanie
				</NavButton>
			</ul>
		)
	}

	render() {
		const { isUserAuth } = this.props

		return (
			<nav className='navigation'>
				<img src={logo} className='logo' alt='logo' />
				{isUserAuth ? this.createAuthNavigation() : this.createNavigation()}
				<button
					id='menuToggleButton'
					onClick={this.onClickDrawerToggle}
					className={`navigation_menu_button ${iconAssets.menu}`}
				/>
			</nav>
		)
	}
}

function mapStateToProps(state) {
	return {
		isUserAuth: state.user.isUserAuth,
	}
}

export default connect(mapStateToProps)(Navbar)
