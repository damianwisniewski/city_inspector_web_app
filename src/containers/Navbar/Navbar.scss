@import '../../assets/styleHelpers/index';

.navigation {
	display: flex;
	align-items: center;
	width: 100%;
	height: 60px;

	@include tablet-size {
		height: 50px;
	}

	&__container {
		position: fixed;
		top: 0;
		left: 0;
		z-index: 500;

		display: flex;
		align-items: center;
		justify-content: space-between;

		width: 100%;
		height: 60px;

		font-size: 14px;

		font-family: $secondary_font;

		background-color: $primary_blue;
		box-shadow: $shadow;

		@include tablet-size {
			height: 50px;
			padding-right: 15px;
		}
	}

	&_list {
		display: flex;
		flex-direction: row;
		justify-content: flex-end;

		margin: 0;
		padding: 5px;

		list-style-type: none;

		@include tablet-size {
			position: absolute;

			top: 50px;
			right: -320px;
			z-index: 500;

			flex-direction: column;
			justify-content: center;

			width: 240px;
			height: calc(100vh - 50px);

			overflow-y: auto;

			background-color: $primary_dark_blue;
			box-shadow: $shadow;

			transition: all 0.3s linear;

			&.opened {
				right: 0;
			}
		}

		& .list_item {
			display: flex;
			align-items: center;

			& > button {
				width: 100%;

				@include tablet-size {
					margin-top: 10px;
					margin-bottom: 10px;

					box-shadow: none;
				}
			}
		}
	}

	&_menu_button {
		display: none;

		&::before {
			padding: 5px;
		}

		@include tablet-size {
			display: block;

			color: $primary_light_blue;

			background-color: transparent;
			border: 2px solid;
			border-radius: 5px;

			outline: none;

			cursor: pointer;

			transition: all 0.2s ease-out;

			&:hover {
				color: $white;
			}
		}
	}

	.logo {
		margin-left: 10px;
	}
}
