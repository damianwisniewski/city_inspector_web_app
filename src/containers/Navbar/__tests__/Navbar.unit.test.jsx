import React from 'react'
import { configure, shallow, mount, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { HashRouter } from 'react-router-dom'
import toJson from 'enzyme-to-json'

import { ModalRouter } from '../../../components/ModalComponents/ModalRouting'
import Navbar from '../Navbar'

configure({ adapter: new Adapter() })

describe('<Navbar />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let NavbarComponent

	beforeEach(() => {
		NavbarComponent = shallow(<Navbar.WrappedComponent />)
	})

	test('should set state "isDrawerOpen" to true, after click - menu toggle button', () => {
		NavbarComponent.setState({ isDrawerOpen: false })
		NavbarComponent.find('#menuToggleButton').simulate('click')
		expect(NavbarComponent.state().isDrawerOpen).toBe(true)
	})

	test('should set state "isDrawerOpen" to false, after click NavItem, for case it was true before', () => {
		NavbarComponent = mount(
			<HashRouter>
				<ModalRouter>
					<Navbar.WrappedComponent isUserAuth={true} />
				</ModalRouter>
			</HashRouter>,
		).find('Navbar')

		NavbarComponent.setState({ isDrawerOpen: true })
		NavbarComponent.find('button#helpLink').simulate('click')
		expect(NavbarComponent.state().isDrawerOpen).toBe(false)
	})

	test('should not change state property "isDrawerOpen", for case it has value false', () => {
		NavbarComponent = mount(
			<HashRouter>
				<ModalRouter>
					<Navbar.WrappedComponent isUserAuth={true} />
				</ModalRouter>
			</HashRouter>,
		).find('Navbar')

		NavbarComponent.setState({ isDrawerOpen: false })
		NavbarComponent.find('button#helpLink').simulate('click')
		expect(NavbarComponent.state().isDrawerOpen).toBe(false)
	})

	/**
	 * Snapshot
	 */
	test('(Auth) should contains proper Navigation items', () => {
		NavbarComponent.setProps({ isUserAuth: true })
		expect(toJson(NavbarComponent)).toMatchSnapshot()
	})

	/**
	 * Snapshot
	 */
	test('(Non-Auth) should contains proper Navigation items', () => {
		NavbarComponent.setProps({ isUserAuth: false })
		expect(toJson(NavbarComponent)).toMatchSnapshot()
	})
})
