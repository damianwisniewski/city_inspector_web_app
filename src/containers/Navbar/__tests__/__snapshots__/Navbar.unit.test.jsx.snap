// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> (Auth) should contains proper Navigation items 1`] = `
<nav
  className="navigation"
>
  <img
    alt="logo"
    className="logo"
    src="logo.svg"
  />
  <ul
    className="navigation_list "
    id="navList"
    onClick={[Function]}
  >
    <NavItem
      className=""
      exact={true}
      icon="map"
      id="mapLink"
      to="/"
      type="link"
    >
      Mapa
    </NavItem>
    <NavItem
      className=""
      exact={true}
      icon="plus"
      id="plusLink"
      to="/nowe_zgloszenie"
      type="link"
    >
      Nowe zgłoszenie
    </NavItem>
    <NavItem
      className=""
      exact={true}
      icon="pin"
      id="pinLink"
      to="/twoje_zgloszenia"
      type="link"
    >
      Twoje zgłoszenia
    </NavItem>
    <NavItem
      className=""
      exact={true}
      icon="eye"
      id="eyeLink"
      to="/subskrypcje"
      type="link"
    >
      Subskrypcje
    </NavItem>
    <NavItem
      className=""
      exact={false}
      icon="bell"
      id="bellLink"
      to="Settings"
      type="button"
    >
      Powiadomienia
    </NavItem>
    <NavItem
      className=""
      exact={true}
      icon="cogs"
      id="cogsLink"
      to="/ustawienia"
      type="link"
    >
      Ustawienia
    </NavItem>
    <NavItem
      className=""
      exact={false}
      icon="logout"
      id="logoutLink"
      to="LogoutPopup"
      type="button"
    >
      Wyloguj
    </NavItem>
  </ul>
  <button
    className="navigation_menu_button menu"
    id="menuToggleButton"
    onClick={[Function]}
  />
</nav>
`;

exports[`<Navbar /> (Non-Auth) should contains proper Navigation items 1`] = `
<nav
  className="navigation"
>
  <img
    alt="logo"
    className="logo"
    src="logo.svg"
  />
  <ul
    className="navigation_list "
    id="navList"
    onClick={[Function]}
  >
    <NavItem
      className=""
      exact={true}
      icon="map"
      id="mapLink"
      to="/"
      type="link"
    >
      Mapa
    </NavItem>
    <NavItem
      className=""
      exact={false}
      icon="help"
      id="helpLink"
      to="HelpPopup"
      type="button"
    >
      Pomoc
    </NavItem>
    <NavButton
      color="white"
      id="registerButton"
      to="SignupPopup"
    >
      Rejestracja
    </NavButton>
    <NavButton
      color="blue"
      id="loginButton"
      to="LoginPopup"
    >
      Logowanie
    </NavButton>
  </ul>
  <button
    className="navigation_menu_button menu"
    id="menuToggleButton"
    onClick={[Function]}
  />
</nav>
`;
